### PR TITLE
Grad record scale

### PR DESCRIPTION
**Pull request by jacenfox**
It's a bug fixed version.

### CODE REVIEW

**Review by MathGaron — COMMENTED** [review]
This is already implemented in the framework. Check the cat_dog exemple : 
in your network init : 
Basically you pass a lambda function that takes the gradient as input and outputs whatever
and pass a tag name so you can retrieve it (fc1 in this case)
`self.hook_fc1 = self.hook_generator(lambda x: float(torch.sqrt(torch.sum(x ** 2))), "fc1")`
in your network forward : 
`if self.training:
x.register_hook(self.hook_fc1)`
and to retrieve the gradient in your callback:
`grad_log = state.model.grad_data["fc1"]`

I am sorry about that, I should really take the time to document it properly....
